### DISCUSSION TITLE
What tools or environment do you use to debug a kit script?

### DISCUSSION THREAD

**Discussion by Claire** [johnlindquist/kit#215] [Q&A]
I'd like to step thru a script I wrote to debug it but I'm not sure how to get the kit environment to run in debug in intellij or vscode. It's  ok for plain js scripts but not where there is use of `args` and `show` internal kit modules
Are we limited to console.logs for now?

**Comment by johnlindquist** [comment]
Great question, I hope to soon have a better answer 😅

Short answer: It's limited for now

I use the helper `inspect` a lot for "debugging" data responses and such, but `console.log` otherwise. Because I'm using the Kit app in dev mode, I get to see a lot more of what's going on.

Longer answer:

If you look at the files in the `~/.kenv/bin` directory, you'll see how node is launched with environment variables and how it points to the scripts in your `~/.kenv/scripts` directory. These files in `~/.kenv/bin` can be run from the terminal since they're technically bash scripts.

The Kit app uses the exact same concept, but with [fork](https://github.com/johnlindquist/kitapp/blob/beta/src/kit.ts#L231) so that an ipc channel opens to send data between the app and node. When you `await arg()` in a script the ipc channel waits for a value from the prompt to send back to the script.

So in the end, it's "just node" and "just a script" with a channel to get values from the app. Even the main menu is a script.

But, since it's the App that launches node (with `fork`) which launches a script, I need to find a way for the App to launch the debugger when it launches node so you can step through each line. I'll start digging in and seeing what's possible.
